hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/spi/meta/impl/InvalidTransactionMetadata.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.meta.impl;
+
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.Transaction;
+import java.util.Collections;
+import java.util.List;
+
+/** Transaction metadata returned in case of any failure during pre-handle. */
+public class InvalidTransactionMetadata implements TransactionMetadata {
```

Comment:
I don't understand why we need this class? If the pre-handle fails, I want to have all the information that went into the pre-handle failing. Then during handle, I can check to see if any of the inputs have changed. If they have not changed, then the result will be the same, and rather than recomputing the result, I can just use the result. Without the inputs, I cannot do this.

For example, suppose I failed in signature verification in prehandle. I cannot just fail the transaction, all I can do is record "with these 3 keys and this transaction data I failed to verify". Then when we get to handle, we can check and see "OK, have any of these 3 keys changed? If yes, I need to do signature verification again. If not, then I can safely report the sig check as having failed".

So it doesn't seem like this class should exist.

Reply:
This class is used when there is an `InvalidProtocolBufferException` while extracting `TransactionBody` . Since the body will not change in pre-handle or handle, I thought we could fail early in handle.